prometheus/src/utils.rs

```diff
@@ -83,6 +83,17 @@ impl<'a> Metric<'a> {
         self.labels.push((label_key, label_value));
         self
     }
+
+    pub fn with_optional_label(
+        self,
+        label_key: &'a str,
+        label_value: Option<String>,
+    ) -> Metric<'a> {
+        if let Some(label_value) = label_value {
+            return self.with_label(label_key, label_value);
+        }
+        self
```

Comment:
```suggestion
        match label_value {
            Some(value) => self.with_label(label_key, value),
            None => self,
        }
```
Can also be written like this, but it’s not shorter so I have no strong opinion on it.